src/modules/messages/messages.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common';
+import { DatabaseService } from 'src/modules/database/database.service';
+import { Prisma } from '@prisma/client';
+
+@Injectable()
+export class MessagesService {
+  constructor(private readonly databaseService: DatabaseService){}
+  
+  async create(createMessageDto: Prisma.MessageCreateInput) {
+    return this.databaseService.message.create({data: createMessageDto});
+  }
+
+  findAll() {
+    return this.databaseService.message.findMany({});
+  }
+
+  findOne(uuid: string) {
+    return this.databaseService.message.findUnique({
+      where: {
+        uuid,
+      },
+    });
+  }
+
+  async update(uuid: string, updateMessageDto: Prisma.MessageUpdateInput) {
+    return this.databaseService.message.update({
+      where: { uuid },
+      data: updateMessageDto,
+    });
+  }
+
+  async remove(uuid: string) {
+    return this.databaseService.message.delete({
```

Comment:
- I suggest if we do only soft delete
- make sure users wont delete messages which they dont have access to